### PR TITLE
Adds JSON-LD functions

### LINKED ISSUE #39: Add support for reading application/ld+json tag
https://developers.google.com/schemas/formats/json-ld

### PR DESCRIPTION
**Pull request by ethanlee16**
Closes #39, [T148837](https://phabricator.wikimedia.org/T148837).

### CODE REVIEW

**Comment by ethanlee16** [comment]
Looks like two of the checks failed due to HTTP error retrieving https://www.uber.com/en-GB - could we write a script to pull in snapshots of websites to store in `/test` instead of fetching for each test? It might make testing more deterministic and reliable, avoiding issues like this.

File: lib/index.js [hunk 1]
```diff
@@ -545,6 +545,33 @@ exports.parseTwitter = BBPromise.method(function(chtml) {
...
+	if (jsonLd.length === 0) {
+		throw new Error("No JSON-LD script tag present on page");
+	}
+
+	jsonLd.each(function() {
+		var contents = chtml(this).text().trim();
+		try {
+			contents = JSON.parse(contents);
+		} catch (e) {
+			throw new Error("Could not parse JSON-LD: " + e);
```
Comment (mvolz) [inline lib/index.js]: What happens if one tag has good data, but one isn't parse-able?

File: index.js [hunk 2]
```diff
@@ -155,6 +155,17 @@ exports.parseTwitter = function(chtml, callback){
 };
 
 /**
+ * Retrieves JSON-LD for given html object
+ *
+ * @param  {Object}   chtml      html Cheerio object
+ * @param  {Function} [callback] optional callback function
+ * @return {Object}              BBPromise for JSON-LD
+ */
+exports.parseJsonLd = function(chtml, callback){
```
Comment (mvolz) [inline index.js]: alphabetical order

File: lib/index.js [hunk 3]
```diff
@@ -545,6 +545,33 @@ exports.parseTwitter = BBPromise.method(function(chtml) {
 
 
 /**
+ * Returns JSON-LD provided by page given HTML object
+ * @param  {Object}   chtml html Cheerio object
+ * @return {Object}   BBPromise for JSON-LD
+ */
+exports.parseJsonLd = BBPromise.method(function(chtml) {
```
Comment (mvolz) [inline lib/index.js]: order

**Comment by ethanlee16** [comment]
The function now fails silently on malformed JSON, only throwing if there are no _valid_ JSON-LD tags on the page. The failing check above is also a result of a timeout, this time as a result of a different test.
¯\_(ツ)_/¯

**Comment by mvolz** [comment]
Whoops, forgot to mention that you should also tick the version number up in package.json as per http://semver.org/

**Comment by mvolz** [comment]
Looks like the Uber ones are still timing out, probably we should just remove those?

File: test/static/turtle_article.json [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+				"iphone": "turtle://",
+				"googleplay": "turtle://"
+			},
+			"id": {
+				"iphone": "000",
+				"googleplay": "superturtlearticle.androidapp"
+			}
+		}
+	}
+}
```
Comment (mvolz) [inline test/static/turtle_article.json]: needs newline at end

File: test/static/turtle_movie.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            "url": "http://www.example.com/turtles/player",
+            "width": "400",
+            "height": "400",
+            "stream": {
+                "url": "http://www.example.com/turtles/turtle.mp4",
+                "content_type": "video/mp4"
+            }
+        }
+    }
+}
```
Comment (mvolz) [inline test/static/turtle_movie.json]: newline

File: test/static.js [hunk 6]
```diff
@@ -17,23 +17,23 @@ describe('static files', function() {
...
+		expected = JSON.parse(fs.readFileSync('./test/static/turtle_movie.json'));
```
Comment (mvolz) [inline test/static.js]: this is much better, thanks!

**Comment by ethanlee16** [comment]
Done - replaced Uber with Apple and it's passing tests! 😄

**Pull request merged**